### PR TITLE
fix: Check ledger range in every handler

### LINKED ISSUE #1565: Check ledger range in every handler
## Issue Description

Ledger range is not checked for having a value inside optional which could lead to undefined behaviour.
For example #1564 fixing UB in a test. It is better to check range and throw an exception (or add assert) in each handler.

### PR DESCRIPTION
**Pull request by PeterChen13579**
fixes #1565

### CODE REVIEW

File: tests/unit/rpc/handlers/AllHandlerTests.cpp [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+    }
+
+    HandlerType handler_;
+
+private:
+    HandlerType
+    initHandler()
+    {
+        if constexpr (std::is_same_v<HandlerType, AccountInfoHandler> || std::is_same_v<HandlerType, FeatureHandler>) {
+            return HandlerType{this->backend, this->mockAmendmentCenterPtr};
```
Comment (kuznetsss) [inline tests/unit/rpc/handlers/AllHandlerTests.cpp:139]: Don't you need to initialise `mockAmendmentCenterPtr` before providing it into the handler here? Looks like it is `nullptr` now, but maybe I'm wrong.

File: tests/unit/rpc/handlers/AllHandlerTests.cpp [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+TYPED_TEST(AllHandlersDeathTest, NoRangeAvailable)
+{
+    // doesn't work without 'this'
+    this->runSpawn(
+        [&](boost::asio::yield_context yield) {
+            TypeParam handler = this->handler_;
+
+            auto const input = createInput<TypeParam>();
+            auto const context = Context{yield};
+            EXPECT_DEATH({ [[maybe_unused]] auto _unused = handler.process(input, context); }, ".*");
```
Comment (kuznetsss) [inline tests/unit/rpc/handlers/AllHandlerTests.cpp]: Maybe you could add a pattern `Assertion .* failed.*`.

File: tests/unit/rpc/handlers/AllHandlerTests.cpp [hunk 3]
```diff
@@ -105,6 +105,7 @@ struct AllHandlersDeathTest : HandlerBaseTest, testing::WithParamInterface<std::
     {
     }
 
+    StrictMockAmendmentCenterSharedPtr mockAmendmentCenterPtr;
```
Comment (kuznetsss) [inline tests/unit/rpc/handlers/AllHandlerTests.cpp:131]: Maybe you could add an ASSERT in `initHandler()` to prevent the error in the future.

**Review by cindyyan317 — CHANGES_REQUESTED** [review]
It's not **every handler.** I did not see `subscribe`  `unsubscribe`, they are immune?

**Review by cindyyan317 — APPROVED** [review]

**Pull request merged**